src/ui/widget-forms/ContentsQueryForm.js

```diff
@@ -0,0 +1,443 @@
+                      {renderLinkTextFields}
+                    </div>
+                  </Collapse>
+                </fieldset>
+              </Col>
+            </Row>
+            <Row>
+              <Col xs={12}>
+                <fieldset className="no-padding">
+                  <FormSectionTitle
+                    titleId="widget.form.frontendFilters"
+                    requireFields={false}
+                    collapsable
+                    onClick={handleCollapseFrontendFilters}
+                  />
+                  <Collapse isOpened={frontendFiltersPanel}>
+                    <FormGroup className="clearfix">
+                      <label htmlFor="categories" className="col-xs-2 control-label">
+                        <FormLabel labelId="app.filter" />
+                      </label>
+                      <Col xs={12} sm={10}>
+                        <FieldArray
+                          intl={intl}
+                          component={FiltersSelectRenderer}
+                          name="userFilters"
+                          options={frontendFilters}
+                          suboptions={frontendFiltersSuboptions}
+                          onResetFilterOption={onResetFilterOption}
+                          filterName="frontendFilters"
+                        />
+                      </Col>
+                    </FormGroup>
+                  </Collapse>
+                </fieldset>
+              </Col>
+            </Row>
+          </div>
+          <br />
+          <Row>
+            <Col xs={12}>
+              {renderSaveButton}
+            </Col>
+          </Row>
+        </form>
+      </Fragment>
+    );
+  }
+}
+
+ContentsQueryFormBody.propTypes = {
+  intl: intlShape.isRequired,
+  language: PropTypes.string.isRequired,
+  languages: PropTypes.arrayOf(PropTypes.shape({})),
+  onDidMount: PropTypes.func.isRequired,
+  handleSubmit: PropTypes.func.isRequired,
+  invalid: PropTypes.bool,
+  submitting: PropTypes.bool,
+  contentTypes: PropTypes.arrayOf(PropTypes.shape({})),
+  contentModels: PropTypes.arrayOf(PropTypes.shape({})),
+  pages: PropTypes.arrayOf(PropTypes.shape({})),
+  categories: PropTypes.arrayOf(PropTypes.shape({})),
+  selectedCategories: PropTypes.arrayOf(PropTypes.string),
+  onResetFilterOption: PropTypes.func.isRequired,
+  onChangeContentType: PropTypes.func.isRequired,
+  onToggleInclusiveOr: PropTypes.func.isRequired,
+  onResetModelId: PropTypes.func.isRequired,
+  selectedContentType: PropTypes.string,
+  selectedInclusiveOr: PropTypes.string,
+};
+
+ContentsQueryFormBody.defaultProps = {
+  invalid: false,
+  submitting: false,
+  languages: [],
+  contentTypes: [],
+  contentModels: [],
+  categories: [],
+  pages: [],
+  selectedCategories: [],
+  selectedContentType: '',
+  selectedInclusiveOr: '',
+};
+
+const ContentsQueryForm = reduxForm({
+  form: 'widgets.contentsQuery',
+})(ContentsQueryFormBody);
+
+export default ContentsQueryForm;
```

Comment:
any clue about how to split this component? it's really big

Reply:
I will split then the whole component into 3 little ones.